odoo/fields.py

```diff
@@ -3485,12 +3485,14 @@ def _add_default_values(self, env, values):
         if not properties_definition:
             return properties_values
 
-        assert isinstance(properties_values, (list, dict))
+        assert isinstance(properties_values, (list, dict, str))
         if isinstance(properties_values, list):
             self._remove_display_name(properties_values)
             properties_list_values = properties_values
-        else:
+        elif isinstance(properties_values, dict):
             properties_list_values = self._dict_to_list(properties_values, properties_definition)
+        else:
+            properties_list_values = self._dict_to_list(ast.literal_eval(properties_values), properties_definition)
```

Comment:
if no, I would move the string check before the dict / list check (so the literal_eval should be line 3488